pkg/plugin/volume_snapshotter_plugin.go

```diff
@@ -43,15 +45,27 @@ var _ velero.VolumeSnapshotter = (*NewVolumeSnapshotter)(nil)
 func (p *NewVolumeSnapshotter) Init(config map[string]string) error {
 	p.Infof("Init called with config: %v", config)
 	p.config = config
-
+	var err error
+	clusterFlavor, err := utils.GetClusterFlavor(nil)
+	if err != nil {
+		p.WithError(err).Errorf("Failed to determine cluster flavour")
+		return err
+	}
+	p.clusterFlavor = clusterFlavor
+	p.Infof("VolumeSnapshotter: Detected cluster flavor: %s", clusterFlavor)
+	// Currently backward compatibility is supported only on vanilla.
+	if clusterFlavor != constants.VSphere {
```

Comment:
Can we change the constant of cluster flavor to `Vanilla`? `VSphere` looks weird.

Reply:
Not this change, that touches way too many files, rebase will become a nightmare. Will do it in a different change.